spec/features/user_can_correct_submission_with_no_business_spec.rb

```diff
@@ -0,0 +1,111 @@
+require 'rails_helper'
+
+RSpec.feature 'Correcting a submission by reporting no business' do
+  before do
+    mock_sso_with(email: 'email@example.com')
+    mock_incomplete_tasks_endpoint!
+    mock_complete_tasks_endpoint!
+    mock_task_with_framework_endpoint!
+    mock_no_business_correction_endpoint!
+    mock_submission_completed_no_business_endpoint!
+  end
+
+  context 'when the latest submission is a business return' do
+    before do
+      mock_completed_task_endpoint!
+    end
+
+    scenario 'user corrects a submission, reporting "no business" when linked to one supplier' do
+      cached_correction_returns_enabled = ENV['CORRECTION_RETURNS_ENABLED']
+      ENV['CORRECTION_RETURNS_ENABLED'] = 'yes'
+
+      mock_user_endpoint!
+
+      visit '/'
+      click_link 'sign-in'
+
+      click_link 'History'
+      click_link 'View'
+      click_link 'Correct this return'
+
+      expect(page).to have_content 'Cheese Board 5'
+      expect(page).to have_content 'July 2018'
+      expect(page).to_not have_content 'Bobs Cheese Shop'
+
+      click_on 'Report no business'
+
+      expect(page).to have_content 'This is a correction'
+      expect(page).to have_content 'Report no business for CBOARD5'
+      expect(page).to_not have_content 'Bobs Cheese Shop'
+
+      click_on 'Confirm no business'
+
+      expect(page).to have_content 'This is a correction'
+      expect(page).to have_content 'CBOARD5'
+      expect(page).to have_content 'July 2018'
+      expect(page).to have_content 'Thank you for reporting no business for'
+      expect(page).to_not have_content 'Bobs Cheese Shop'
+
+      ENV['CORRECTION_RETURNS_ENABLED'] = cached_correction_returns_enabled
+    end
+
+    scenario 'user can see the current supplier name on the complete no business
+      views when linked to multiple suppliers' do
+      cached_correction_returns_enabled = ENV['CORRECTION_RETURNS_ENABLED']
+      ENV['CORRECTION_RETURNS_ENABLED'] = 'yes'
+
+      mock_user_with_multiple_suppliers_endpoint!
+
+      visit '/'
+      click_link 'sign-in'
+
+      click_link 'History'
+      click_link 'View'
+      click_link 'Correct this return'
+
+      expect(page).to have_content 'Cheese Board 5'
+      expect(page).to have_content 'July 2018'
+      expect(page).to have_content 'Bobs Cheese Shop'
+
+      click_on 'Report no business'
+
+      expect(page).to have_content 'This is a correction'
+      expect(page).to have_content 'Report no business for CBOARD5'
+      expect(page).to have_content 'Bobs Cheese Shop'
+
+      click_on 'Confirm no business'
+
+      expect(page).to have_content 'This is a correction'
+      expect(page).to have_content 'CBOARD5'
+      expect(page).to have_content 'July 2018'
+      expect(page).to have_content 'Thank you for reporting no business for'
+      expect(page).to have_content 'Bobs Cheese Shop'
+
+      ENV['CORRECTION_RETURNS_ENABLED'] = cached_correction_returns_enabled
```

Comment:
as above, either use `before`/`after` or `ensure` so that the env variable is always reset.

Reply:
Rather than that, I went the whole hog and added `climate_control` - we already use this gem for the API, so it's familiar to the project, and it's much cleaner and understandable than mucking about with the `ENV` directly.